addon_imps/storage/dropbox.py

```diff
@@ -0,0 +1,113 @@
+import dataclasses
+import typing
+
+from addon_toolkit.interfaces import storage
+
+
+class DropboxStorageImp(storage.StorageAddonHttpRequestorImp):
+    """storage on dropbox.com
+    see https://www.dropbox.com/developers/documentation/http/documentation
+    """
+
+    async def get_external_account_id(self, _: dict[str, str]) -> str:
+        return ""
+
+    async def list_root_items(self, page_cursor: str = "") -> storage.ItemSampleResult:
+        if page_cursor:
+            async with self.network.POST(
+                "/files/list_folder/continue",
+                {"cursor": page_cursor},
+            ) as _response:
+                _parsed = _DropboxParsedJson(await _response.json_content())
+                return storage.ItemSampleResult(
+                    items=list(_parsed.item_results()),
+                )
+
+        return await self.get_item_info("")
+
+    async def get_item_info(self, item_id: str) -> storage.ItemSampleResult:
+        if not item_id:
+            async with self.network.POST(
+                "files/list_folder",
+                {
+                    "include_non_downloadable_files": True,
+                    "path": "/",
+                    "recursive": True,
```

Comment:
what does recursive parameter mean? Would that return contents of subfolders? I don't believe this is the behaviour we'd want

Reply:
Yes, the recursive parameter means that returns the contents of subfolder. I thought that we need it, I'll remove that behaviour.